src/ipc/ipc4/helper.c

```diff
@@ -652,11 +653,16 @@ __cold int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 	}
 
 	/* these might call comp_ipc4_bind_remote() if necessary */
-	ret = comp_bind(source, bu);
+	bind_data.ipc4_data = bu;
+	bind_data.as_source = NULL;
+	bind_data.as_sink = audio_buffer_get_sink(&buffer->audio_buffer);
+	ret = comp_bind(source, &bind_data);
 	if (ret < 0)
 		goto e_src_bind;
 
-	ret = comp_bind(sink, bu);
+	bind_data.as_sink = NULL;
+	bind_data.as_source = audio_buffer_get_source(&buffer->audio_buffer);
+	ret = comp_bind(sink, &bind_data);
```

Comment:
This is a bit awkward with setting either as_sink or as_source to NULL. I wonder if this could be more readable with a helper macro to create bind_data, or a helper macro to call comp_bind for an audio sink/source.

On related note, this ipc_comp_connect() is getting out of hand complexity wise and would benefit having some of the functionality being put to subroutines to make this more readable.